core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -677,9 +701,29 @@ private List<ServiceConfigSetters> getServiceConfigSetters(
         } else {
             serviceConfigSetters = ImmutableList.copyOf(this.serviceConfigSetters);
         }
+
+        if (defaultVirtualHostBuilder.contextPath() != null) {
+            return applyPrefixToServiceConfigSetters(serviceConfigSetters,
+                                                     defaultVirtualHostBuilder.contextPath());
+        }
+
         return serviceConfigSetters;
     }
 
+    private List<ServiceConfigSetters> applyPrefixToServiceConfigSetters(
+            List<ServiceConfigSetters> serviceConfigSetters, String contextPath) {
+        final List<ServiceConfigSetters> prefixedConfigSetters = new ArrayList<>(serviceConfigSetters.size());
+        for (ServiceConfigSetters setters : serviceConfigSetters) {
+            if (setters instanceof ServiceConfigBuilder) {
+                prefixedConfigSetters.add(((ServiceConfigBuilder) setters)
+                                                  .getPrefixedServiceConfigBuilder(contextPath));
+            } else {
+                prefixedConfigSetters.add(setters);
+            }
+        }
+        return prefixedConfigSetters;
+    }
```

Comment:
I'm wondering if there's no problem that prefixing `contextPath` to `Route` when building `VitrualHost`. 🤔 
In this way, I didn't need to modify findVirtualHost().
I'd appreciate it if you check this approach has side effects or any concerns.. 😓

Reply:
I agree the current approach sounds better 👍